integration_tests/helpers/constrainedMiningNodeProcess.js

```diff
@@ -0,0 +1,101 @@
+const {getFreePort} = require("./util");
+const dateFormat = require('dateformat');
+const fs = require('fs');
+const {spawnSync, spawn, execSync} = require('child_process');
+const {expect} = require('chai');
+const {createEnv} = require("./config");
+
+let outputProcess;
+
+class ConstrainedMiningNodeProcess {
+    constructor(name, baseNodeAddress, walletAddress ) {
+        this.name = name;
+        this.maxBlocks = 1;
+        this.minDiff = 10000;
+        this.maxDiff = 100000;
+        this.nodeAddress = baseNodeAddress.split(":")[0];
+        this.nodeGrpcPort = baseNodeAddress.split(":")[1];
+        this.walletAddress = walletAddress.split(":")[0];
+        this.walletGrpcPort = walletAddress.split(":")[1];
+    }
+
+    async init(maxBlocks, minDiff, maxDiff) {
+        this.maxBlocks = maxBlocks;
```

Comment:
Add a default so that these don't have to be specified

```suggestion
        this.maxBlocks = maxBlocks || this.maxBlocks;
```